submariner-operator/issue-79-kubeconfig-handling.md

```diff
@@ -0,0 +1,122 @@
+subctl foo --kubeconfig /path/to/kubeconfig --kubecontext bar
+subctl --kubeconfig /path/to/kubeconfig --kubecontext bar foo
+```
+
+For commands using multiple contexts, the prefixes should make the purpose of each context clear,
+using appropriate prefixes when setting up `clientcmd`:
+
+```bash
+KUBECONFIG=/path/to/kubeconfig subctl benchmark latency --fromcontext foo --tocontext bar
+subctl benchmark latency --kubeconfig /path/to/kubeconfig --fromcontext foo --tocontext bar
+```
+
+`--kubeconfig` is preserved as-is, since users can use a single file combining all their contexts.
+However such commands should also support separate kubeconfigs, to allow using kubeconfigs which
+contain conflicting configuration (_e.g._ kubeconfigs using the same context name and user name,
+as produced by the OpenShift installer):
+
+```bash
+subctl benchmark latency --fromconfig /path/to/fromconfig --toconfig /path/to/toconfig
+```
+
+In such scenarios, if one “named” kubeconfig flag is used, all corresponding kubeconfigs must be
+specified, without fallback to a generic `--kubeconfig` flag or the `KUBECONFIG` environment
+variable.
+
+For commands potentially using _all_ accessible contexts (`subctl gather`), the existing behaviour
+should be preserved:
+
+```bash
+# Use all accessible contexts
+KUBECONFIG=/path/to/kubeconfig subctl gather
+subctl gather --kubeconfig /path/to/kubeconfig
+subctl gather /path/to/kubeconfig
+
+# Use named contexts
+KUBECONFIG=/path/to/kubeconfig subctl gather --kubecontexts context1,context2
+subctl gather --kubeconfig /path/to/kubeconfig --kubecontexts context1,context2
+subctl gather /path/to/kubeconfig --kubecontexts context1,context2
+```
+
+`clientcmd` _doesn’t_ handle `--kubeconfig` itself, that needs to be handled by `subctl`.
+
+This enhancement should also include a test suite covering all the different possibilities.
+As far as possible, the 0.12 options should keep their existing behaviour, and be marked for deprecation;
+they will be removed after 0.13.
+
+This would subsume [`--kubecontexts` support in `subctl diagnose`](https://github.com/submariner-io/submariner-operator/issues/1327).
+
+## Definition of Done (Checklist)
+<!-- Make sure to check all relevant items before end of the release: -->
+
+* [ ] Code complete
+* [ ] Relevant metrics added
+* [ ] The acceptance criteria met
+* [ ] Unit/e2e test added & pass
+* [ ] CI jobs pass
+* [ ] Deployed using cloud-prepare+subctl
+* [ ] Run subctl verify, diagnose and gather
+* [ ] Uninstall
+* [ ] Troubleshooting (gather/diagnose) added
+* [ ] Documentation added
+* [ ] Release notes added
+
+## Work Items
+
+* [ ] Create a test suite for the various `--kubeconfig` and `--kubecontext` variants
+* [ ] Delegate flag setup for `--kubecontext` and related flags to `k8s.io/client-go/tools/clientcmd`
+* [ ] Integrate clientcmd handling into `restconfig.Producer`
+* [ ] Where appropriate, add `--kube` variants for command which need multiple contexts
```

Comment:
I don't think these are needed in the EP as they are part of the epic issue template.